src/services/Store/StoreProvider.tsx

```diff
@@ -240,11 +236,7 @@ export const StoreProvider: React.FC = ({ children }) => {
 
   useEffect(() => {
     if (mainnetAccounts.length > 0) {
-      HistoryService.instance.getHistory(mainnetAccounts).then((history) => {
-        if (history !== null) {
-          setTxHistory(history);
-        }
-      });
+      dispatch(fetchHistory());
```

Comment:
Can we move this conditional in the saga and dispatch the event at `handleRehydrateSuccess` in `persistence.slice.ts` ?

Reply:
We need it to run every time the accounts change though. So that is not enough.

Reply:
Possibly. Will look into this.